File: lambda/email-processor/send_email.py
import os
import json
import boto3


REGION = os.getenv("AWS_REGION", "us-east-2")
SENDER_EMAIL = os.environ["SENDER_EMAIL"]
RECIPIENT_EMAIL = os.environ["RECIPIENT_EMAIL"]

ses = boto3.client("ses", region_name=REGION)


def handler(event, context):
    """
    Expected event shape:
    {
      "subject": "Subject line",
      "body":    "Plain-text email body"
    }
    """

    subject = event.get("subject", "Hello from Lambda")
    body = event.get("body", "This email was sent via Amazon SES!")

    try:
        resp = ses.send_email(
            Source=SENDER_EMAIL,
            Destination={"ToAddresses": [RECIPIENT_EMAIL]},
            Message={
                "Subject": {"Data": subject, "Charset": "UTF-8"},
                "Body": {
                    "Text": {"Data": body, "Charset": "UTF-8"},
                    # uncomment to send HTML, too:
                    # "Html": {"Data": f"<p>{body}</p>", "Charset": "UTF-8"},
                },
            },
        )

        return {
            "statusCode": 200,
            "body": json.dumps(
                {"message": "Email sent", "messageId": resp["MessageId"]}
            ),
        }

    except ses.exceptions.MessageRejected as err:
        # Most common sandbox / verification error
        return {"statusCode": 400, "body": json.dumps({"error": str(err)})}

    except Exception as err:
        return {"statusCode": 500, "body": json.dumps({"error": str(err)})}

File: lambda/sendEmail.ts
import { AppSyncResolverHandler } from "aws-lambda";
import { SESClient, SendEmailCommand } from "@aws-sdk/client-ses";
import { logger, metrics, tracer } from "../powertools/utilities";
import { SendEmailInput } from "../appsync";

export const handler: AppSyncResolverHandler<
  { input: SendEmailInput },
  Boolean
> = async (event, _context) => {
  const REGION = process.env.AWS_REGION ?? "us-east-2";
  const ses = new SESClient({ region: REGION });

  logger.info(`received event is ${event.arguments.input}`);

  try {
    const { MessageId } = await ses.send(
      new SendEmailCommand({
        Source: event.arguments.input.from,
        Destination: { ToAddresses: event.arguments.input.to },
        Message: {
          Subject: { Data: event.arguments.input.subject!, Charset: "UTF-8" },
          Body: {
            Text: { Data: event.arguments.input.plainBody!, Charset: "UTF-8" },
            // Html: { Data: `<p>${body}</p>`, Charset: "UTF-8" }, // optional
          },
        },
      })
    );

    logger.info(`Email sent with messageId: ${MessageId}`);

    return true;
  } catch (err: any) {
    logger.info(`Email failed to send with err: ${err}`);
    return false;
  }
};
